core/src/main/java/com/linecorp/armeria/server/ServerListenerBuilder.java

```diff
@@ -262,6 +265,62 @@ public ServerListenerBuilder whenStopped(Iterable<? extends Consumer<? super Ser
         return this;
     }
 
+    /**
+     * Add a callback that gracefully shuts down the given {@link ExecutorService} when the {@link Server}
+     * is stopping. It will wait indefinitely for the {@link ExecutorService} to terminate during shutdown.
+     */
+    public ServerListenerBuilder shutdownWhenStopping(ExecutorService executorService) {
+        requireNonNull(executorService, "executorService");
+        serverStoppingCallbacks.add(s -> ShutdownSupport.of(executorService).shutdown());
+        return this;
+    }
+
+    /**
+     * Add a callback that gracefully shuts down the given {@link ExecutorService} when the {@link Server}
+     * is stopping. It allows a maximum duration of {@code terminationTimeout} for the {@link ExecutorService}
+     * to terminate gracefully before it is forcefully terminated.
+     */
+    public ServerListenerBuilder shutdownWhenStopping(ExecutorService executorService,
+                                                      Duration terminationTimeout) {
+        requireNonNull(executorService, "executorService");
+        requireNonNull(terminationTimeout, "terminationTimeout");
+        long nanoSeconds;
+        try {
+            nanoSeconds = terminationTimeout.toNanos();
+        } catch (ArithmeticException ignore) {
+            nanoSeconds = terminationTimeout.isNegative() ? Long.MIN_VALUE : Long.MAX_VALUE;
+        }
+        shutdownWhenStopping(executorService, nanoSeconds, TimeUnit.NANOSECONDS);
+        return this;
+    }
+
+    /**
+     * Add a callback that gracefully shuts down the given {@link ExecutorService} when the {@link Server}
+     * is stopping. It allows a maximum duration of {@code terminationTimeout} in the specified
+     * {@code timeUnit} for the {@link ExecutorService} to terminate gracefully before it is forcefully
+     * terminated.
+     */
+    public ServerListenerBuilder shutdownWhenStopping(ExecutorService executorService, long terminationTimeout,
+                                                      TimeUnit timeUnit) {
```

Comment:
Conventionally, we provide `Duration` and `long` which implies milliseconds for a duration.
https://github.com/line/armeria/blob/a0684e3ea473a2c1ff1ebdf0a2697fae4fdb697d/core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java#L1850-L1859
```suggestion
    public ServerListenerBuilder shutdownWhenStopping(ExecutorService executorService, long terminationTimeoutMillis) {
```

Reply:
I have also made modifications to `ShutdownSupport#of` to accept `terminationTimeoutMillis` as a parameter. Alternatively, should I leave the `ShutdownSupport`, which is only used internally, as it is, to provide greater flexibility in selecting the `TimeUnit`?